packages/request-manager/src/torIdentities.ts

```diff
@@ -0,0 +1,83 @@
+import http from 'http';
+import https from 'https';
+import { SocksProxyAgent } from 'socks-proxy-agent';
+import { RequestInit } from 'node-fetch';
+import { TorController } from './controller';
+
+interface TorRequestOptions {
+    identity?: string;
+}
+
+export class TorIdentities {
+    static torController: TorController;
+    static identities: { [key: string]: SocksProxyAgent };
+
+    public static init(torController: TorController) {
+        this.torController = torController;
+        this.identities = {};
+        // When initializing create default identity, so it is ready to use.
+        this.getIdentity('default');
+    }
+
+    private static getIdentity(identity: string): SocksProxyAgent {
+        if (!this.identities[identity]) {
+            this.identities[identity] = new SocksProxyAgent({
+                hostname: this.torController.options.host,
+                port: this.torController.options.port,
+                userId: identity,
+                password: identity,
+            });
+        }
+        return this.identities[identity];
+    }
+
+    public static fetch(url: string, init: RequestInit, torRequestOptions?: TorRequestOptions) {
```

Comment:
which is better? :thinking: :duck: ? 
`fetch(url: string, init: RequestInit, torRequestOptions?: TorRequestOptions)`
or 
`fetch(url: string, init: RequestInit & TorRequestOptions)`

Reply:
Maybe the second one a tiny bit, but both are pretty good imo as they both allow to replace original `fetch` without knowing about the added functionality.